sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -49,38 +49,30 @@ protected StringComparison CaseSensitivity
 
         protected abstract bool IsStatic(TMemberDeclarationSyntax member);
 
+        protected abstract bool IsAbstract(TMemberDeclarationSyntax member);
+
         protected void CheckMembers(SyntaxNodeAnalysisContext c, IEnumerable<TMemberDeclarationSyntax> members)
         {
-            var misplacedOverloadsMapping = GetMisplacedOverloads(members);
-            foreach (var misplacedOverloadsEntry in misplacedOverloadsMapping)
+            var misplacedOverloads = GetMisplacedOverloads(c, members);
+            foreach (var misplacedMethods in misplacedOverloads.Values.Where(x => x.Count > 1))
```

Comment:
I think the filtering `misplacedOverloads.Values.Where(x => x.Count > 1))` can already be done in the `GetMisplacedOverloads` method, as don't actually use the dictionary key here anyway

Reply:
in addition, maybe use `X.Any()` instead of `x.Count > 1` ?

Reply:
`X.Any()` is for `x.Count <>0`. But `GetMisplacedOverloads` actually adds first item as legal primary location and every other item as illegal location. Only those with count>1 are illegal then.

I've updated the `GetMisplacedOverloads` to do this check as Chrisophe suggested, because it makes much more sense. In old version even well placed where in the collection.